src/SFA.DAS.AANHub.Api.UnitTests/Controllers/AdminControllerTests.cs

```diff
@@ -71,5 +73,105 @@ public AdminControllerTests()
 
             result.As<BadRequestObjectResult>().StatusCode.Should().Be(StatusCodes.Status400BadRequest);
         }
+
+        [Test]
+        [AutoMoqData]
+        public async Task GetAdmin_InvokesQueryHandler(
+            [Frozen] Mock<IMediator> mediatorMock,
+            [Greedy] AdminsController sut,
+            string userName,
+            ValidatedResponse<GetAdminMemberResult> handlerResult)
+        {
+            mediatorMock.Setup(m => m.Send(It.IsAny<GetAdminMemberQuery>(), It.IsAny<CancellationToken>())).ReturnsAsync(handlerResult);
+
+            var response = await sut.GetAdmin(userName);
+            response.Should().NotBeNull();
+
+            var result = response as OkObjectResult;
+            Assert.AreEqual(StatusCodes.Status200OK, result!.StatusCode);
+
+            var queryResult = result.Value as GetAdminMemberResult;
+            queryResult.Should().BeEquivalentTo(handlerResult.Result);
+
+            mediatorMock.Verify(m => m.Send(It.IsAny<GetAdminMemberQuery>(), It.IsAny<CancellationToken>()));
+        }
+
+        [Test]
+        [AutoMoqData]
+        public async Task GetAdmin_InvokesQueryHandler_NoResultGivesNotFound(
+            [Frozen] Mock<IMediator> mediatorMock,
+            [Greedy] AdminsController sut,
+            string userName)
+        {
+            var errorResponse = new ValidatedResponse<GetAdminMemberResult>
+                (new List<ValidationFailure>());
+
+            mediatorMock.Setup(m => m.Send(It.Is<GetAdminMemberQuery>(q => q.UserName == userName), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(errorResponse);
+
+            var response = await sut.GetAdmin(userName);
+
+            var result = response as NotFoundResult;
+            result.Should().NotBeNull();
+            Assert.AreEqual(StatusCodes.Status404NotFound, result!.StatusCode);
+
+            mediatorMock.Verify(m => m.Send(It.IsAny<GetAdminMemberQuery>(), It.IsAny<CancellationToken>()));
+        }
+
+        [Test]
+        [AutoMoqData]
+        public async Task GetAdmin_InvokesQueryHandler_ResultGivesSuccessfulResult(
```

Comment:
This is same as the first test, isn't it?

Reply:
The first test is for the Create endpoint, this is for Get